doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1856 @@
+
+##### 1.2.1.3.1 Local MAC learning on Multihomed ES
+MAC learned dynamically over an interface which is associated with a Ethernet-segment is considered as Local MAC address. Existing local MAC learning mechanism is used for the EVPN MH support and no additional changes are required.
+
+##### 1.2.1.3.2 Local MAC ageing
+FdbOrch agent is notified with age event when local MAC address aged out from the hardware. FdbOrch requires extension to re-program the MAC with L2 NHID if it still has active tunnel members. No changes are required to the current mechanism of MAC address being deleted from STATE_DB in order to notify the Fdbsyncd about the local ageing.
+
+##### 1.2.1.3.3 Remote MAC learning on Multihomed ES
+EVPN VxLAN remote MAC addresses are installed into Linux FDB by FRR and Fdbsyncd updates them into VXLAN_FDB_TABLE. The same VXLAN_FDB_TABLE is extended for installing MAC on the ES when the MAC update is received from EVPN MH ES peer; a new type is added to VXLAN_FDB_TABLE to handle synchronized entry.
+
+**Remote MAC with local ESI presence**: A new "ifname" field is added to VXLAN_FDB_TABLE. The MAC updates received from ES peer is installed with ifname set to the interface associated with an ESI. Fdborch is enhanced to add the EVPN remote MAC against the local interface with aging disabled.
+
+**Remote MAC with no local ESI:** MAC updates from remote EVPN MH is installed in VXLAN_FDB_TABLE with L2 NHID instead of a specific tunnel. FdbOrch is extended to program the MAC address against L2 NHID.
+
+##### 1.2.1.3.4  Remote MAC aging
+When MAC address gets aged on remote VTEP, it sends a withdrawal notification. FRR must do:
+- Remove the MAC learn ownership from the corresponding remote VTEP.
+- Check if the MAC address is learned locally or got advertised from any other remote VTEP nodes.
+- If local or any other remote VTEP advertised the MAC with same ESI, no updates sent to Fdbsyncd/Orchagent. Else the MAC will be removed from VXLAN_FDB_TABLE, and FdbOrch will subsequently remove the MAC from hardware.
+
+##### 1.2.1.3.5  Local Multihomed ES Link down
+Upon access interface going operationally down, FdbOrch gets notified to flush the MAC addresses associated with the interface. Fdborch is extended  to check if the interface is associated with an Ethernet-Segment. If the interface is an orphan port then flush associated MAC addresses. If there is an Ethernet Segment associated with the interface, FdbOrch fetches the corresponding L2 NHID and re-program all (local and remote) MAC addresses pointing to L2 NHID associated with the local ESI. The MAC addresses are programmed with aging disabled.
+
+##### 1.2.1.3.6  Local Multihomed ES Link up
+Upon access interface going operationally up, new support is added in the Orchagent to move the MAC from L2 NHID to the corresponding interface which is associated with the Ethernet Segment. MAC addresses are programmed with aging disabled.
```

Comment:
@skumar041 , if the mac is being programmed locally (after the link comes up,) the only reason is that the mac is active on another VTEP - otherwise, the mac would have been removed from the evpn as none of the VTEPs have it.
In other words, the mac is in control of the remote VTEP, and therefore it should be programmed with ageing disabled on the local VTEP.
If the mac is programmed as ageing enabled, it means there is a local mac learn event - which may not happen after the link comes up.

Reply:
With EVPN control plane, learning / ageing never happen over remote VTEP.